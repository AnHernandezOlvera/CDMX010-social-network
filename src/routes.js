import { home } from './home.js';
import posting from './posting.js';
import { post } from './post.js';
import { singlepost } from './single-post.js';

const rootDiv = document.getElementById('root');

export const routes = {
  '/': home,
  '/posting': posting,
  '/post': post,
  '/singlepost': singlepost,
};

export const onNavigate = (pathname) => {
  window.history.pushState({}, pathname, window.location.origin + pathname);
  const component = routes[pathname];
  component(rootDiv);
};

window.onpopstate = () => {
  const component = routes[window.location.pathn];
  component(rootDiv);
};
