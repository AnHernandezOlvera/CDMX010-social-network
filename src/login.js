import { onNavigate } from './routes.js';

import { access, observer, validacion } from './lib/firebase.js';

export const login = async (target) => {
  const templeteLogin = `
        <div id="loginView">
        <form id="loginForm">
        <h4>Acceso</h4>
        <input type="email" id="emailAccess" placeholder="Email" />
        <input type="password" id="passwordAccess" placeholder="Password" />
        <button id="access">Acceder</button>
        </form>
        </div>
  `;
    // eslint-disable-next-line no-param-reassign
  target.innerHTML = templeteLogin;

  const accessButton = document.getElementById('access');

  accessButton.addEventListener('click', async (e) => {
    e.preventDefault();
    const emailAccess = document.getElementById('emailAccess').value;
    const passwordAccess = document.getElementById('passwordAccess').value;
    access(emailAccess, passwordAccess);
    console.log(emailAccess, passwordAccess);
    const usuario = await validacion(emailAccess);
    console.log(usuario);
    onNavigate('/home');
  });
};
export default login;
