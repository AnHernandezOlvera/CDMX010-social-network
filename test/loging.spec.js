/* eslint-disable max-len */
/* eslint-disable jest/no-disabled-tests */
import { screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';
import { login } from '../src/login.js';
import { onNavigate } from '../src/routes.js';

describe('login', () => {
  beforeEach(() => {
    document.body.innerHTML = '<div id="root"></div>';
  });
  it('should take me to the post when I click the post title', async () => {
    const target = document.getElementById('root');

    // Mockeamos la funcion de login de Firebase la cual sera invocada cuando el usuario haga click sobre el boton "Sign In"
    // Por que?  La idea central es verificar que esta funcion es invocada lo cual haremos linea mas abajo
    const access = jest.fn().mockImplementation(() => Promise.resolve('ok'));
    const firebase = { access };
    await login(target, firebase);
    expect(target.innerHTML).toMatchSnapshot();
    const email = 'test@example.com';
    const password = '123456789';
    // Referencia: https://testing-library.com/docs/ecosystem-user-event/#api
    userEvent.type(screen.getByLabelText('Email'), email);
    userEvent.type(screen.getByLabelText('Password'), password);
    userEvent.click(screen.getByDisplayValue('Acceder'));
    // Verificamos que nuestro mock de la funcion login de firebase es llamada con el email y password que el usuario escribio mas arriba
    // Referencia: https://jestjs.io/docs/en/expect#tohavebeencalledwitharg1-arg2-
    expect(access).toHaveBeenCalledWith(email, password, onNavigate);
  });
});
