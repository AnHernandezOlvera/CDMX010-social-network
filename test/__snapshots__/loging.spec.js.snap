// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login should take me to the post when I click the post title 1`] = `
"
        <div id=\\"loginView\\">
        <form id=\\"loginForm\\">
        <h4>Acceso</h4>
        <label for=\\"emailAccess\\">Email</label>
        <input name=\\"email \\" type=\\"email\\" id=\\"emailAccess\\" placeholder=\\"Email\\">
        <label for=\\"passwordAccess\\">Password</label>
        <input type=\\"password\\" id=\\"passwordAccess\\" placeholder=\\"Password\\">
        <input id=\\"access\\" type=\\"submit\\" value=\\"Acceder\\">
        <button id=\\"accessGoogle\\">Acceder con Google</button>
        </form>
        </div>
  "
`;
